### PR TITLE
feat(locale): Add locale ro_MD

### PR DESCRIPTION
**Pull request by botzill**
Add a new locale for ro_MD

### CODE REVIEW

File: src/locales/ro_MD/date/month.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    'Aprilie',
+    'Mai',
+    'Iunie',
+    'Iulie',
+    'August',
+    'Septembrie',
+    'Octombrie',
+    'Noiembrie',
+    'Decembrie',
+  ],
```
Comment (matthewmayer) [inline src/locales/ro_MD/date/month.ts]: you can omit this as identical

File: src/locales/ro_MD/date/month.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    'Apr',
+    'Mai',
+    'Iun',
+    'Iul',
+    'Aug',
+    'Sep',
+    'Oct',
+    'Noi',
+    'Dec',
+  ],
```
Comment (matthewmayer) [inline src/locales/ro_MD/date/month.ts]: you can omit this as identical

File: src/locales/ro_MD/date/weekday.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  // It is used to specify a word in context, which may differ from a stand-alone word
+  wide_context: [
+    'Luni',
+    'Marți',
+    'Miercuri',
+    'Joi',
+    'Vineri',
+    'Sâmbătă',
+    'Duminică',
+  ],
```
Comment (matthewmayer) [inline src/locales/ro_MD/date/weekday.ts]: you can omit this as identical

File: src/locales/ro_MD/date/weekday.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  // It is used to specify a word in context, which may differ from a stand-alone word
+  abbr_context: [
+    'Luni',
+    'Marți',
+    'Miercuri',
+    'Joi',
+    'Vineri',
+    'Sâmbătă',
+    'Duminică',
+  ],
```
Comment (matthewmayer) [inline src/locales/ro_MD/date/weekday.ts]: you can omit this as identical

File: src/locales/ro_MD/internet/domain_suffix.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export default [
+  'com',
+  'ru',
+  'net',
+  'org',
+  // Moldavian
+  'md',
+  'ro.ru'
```
Comment (matthewmayer) [inline src/locales/ro_MD/internet/domain_suffix.ts]: are the ru and ro.ru domains popular in moldova?

File: src/locales/ro_MD/internet/free_email.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export default ['mail.ru', 'rambler.ru', 'yandex.ru', 'gmail.com', 'gmail.ru', 'yahoo.com', 'hotmail.com'];
```
Comment (matthewmayer) [inline src/locales/ro_MD/internet/free_email.ts]: are .ru email addresses popular in Moldova?

File: src/locales/ro_MD/location/building_number.ts [hunk 7]
```diff
@@ -0,0 +1 @@
+export default ['Bloc ##'];
```
Comment (matthewmayer) [inline src/locales/ro_MD/location/building_number.ts]: are there any other common patterns?

File: src/locales/ro_MD/location/city_name.ts [hunk 8]
```diff
@@ -0,0 +1,96 @@
+export default [
+  'Chișinău',
```
Comment (matthewmayer) [inline src/locales/ro_MD/location/city_name.ts]: please sort alphabetically

File: src/locales/ro_MD/location/street_name.ts [hunk 9]
```diff
@@ -0,0 +1,300 @@
+export default [
+  'Stefan Cel Mare',
```
Comment (matthewmayer) [inline src/locales/ro_MD/location/street_name.ts]: please sort alphabetically

File: src/locales/ro_MD/location/street_suffix.ts [hunk 10]
```diff
@@ -0,0 +1 @@
+export default ['Aleea', 'Bulevardul', 'Intrarea', 'Str-la'];
```
Comment (matthewmayer) [inline src/locales/ro_MD/location/street_suffix.ts]: should be street_prefix

File: src/locales/ro_MD/person/female_first_name.ts [hunk 11]
```diff
@@ -0,0 +1,260 @@
+export default [
+  'Tatiana',
```
Comment (matthewmayer) [inline src/locales/ro_MD/person/female_first_name.ts]: please sort alphabetically

File: src/locales/ro_MD/person/last_name.ts [hunk 12]
```diff
@@ -0,0 +1,301 @@
+export default [
+  'Rusu',
```
Comment (matthewmayer) [inline src/locales/ro_MD/person/last_name.ts]: please sort alphabetically

File: src/locales/ro_MD/person/male_first_name.ts [hunk 13]
```diff
@@ -0,0 +1,217 @@
+export default [
+  'Andrei',
```
Comment (matthewmayer) [inline src/locales/ro_MD/person/male_first_name.ts]: please sort alphabetically

File: src/locales/ro_MD/person/name.ts [hunk 14]
```diff
@@ -0,0 +1,8 @@
+export default [
+  { value: '{{person.firstName}} {{person.lastName}}', weight: 8 },
+  { value: '{{person.lastName}} {{person.firstName}}', weight: 1 },
```
Comment (matthewmayer) [inline src/locales/ro_MD/person/name.ts]: is it common to write lastName firstName? if not, please remove.

File: src/locales/ro_MD/person/suffix.ts [hunk 15]
```diff
@@ -0,0 +1 @@
+export default ['Jr.', 'Sr.'];
```
Comment (matthewmayer) [inline src/locales/ro_MD/person/suffix.ts]: as this is not used in the name patterns, can be removed.

**Review by matthewmayer — CHANGES_REQUESTED** [review]

**Review by matthewmayer — CHANGES_REQUESTED** [review]
Also run `pnpm run test` and check for errors, CI reports two about duplicate entries:
```
⎯⎯⎯⎯⎯⎯⎯ Failed Tests 2 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  test/locales.spec.ts > locale > ro_MD > location > city_name > should not have duplicate entries
Error: Duplicated values are [Florești]

 FAIL  test/locales.spec.ts > locale > ro_MD > person > female_first_name > should not have duplicate entries
Error: Duplicated values are [Ianna, Irina]
```

File: src/locales/ro_MD/date/month.ts (hunk 1 above)
Comment (botzill) [reply src/locales/ro_MD/date/month.ts]: You mean set the array empty and it will take automatically from `ro` locale ?

File: src/locales/ro_MD/internet/domain_suffix.ts (hunk 5 above)
Comment (botzill) [reply src/locales/ro_MD/internet/domain_suffix.ts]: There are some yes

File: src/locales/ro_MD/person/female_first_name.ts (hunk 11 above)
Comment (botzill) [reply src/locales/ro_MD/person/female_first_name.ts]: OK

File: src/locales/ro_MD/person/last_name.ts (hunk 12 above)
Comment (botzill) [reply src/locales/ro_MD/person/last_name.ts]: OK

File: src/locales/ro_MD/date/month.ts (hunk 1 above)
Comment (ST-DDT) [reply src/locales/ro_MD/date/month.ts]: No, don't set the value.
If the entry isn't present then we will automatically fall back to the main one.

File: src/locales/ro_MD/location/street.ts [hunk 16]
```diff
@@ -0,0 +1 @@
+export default ['{{location.street_suffix}} {{location.street_name}}'];
```
Comment (ST-DDT) [inline src/locales/ro_MD/location/street.ts]: A (street) suffix should be at the end. Or be called prefix.

File: src/locales/ro_MD/date/month.ts (hunk 1 above)
Comment (ST-DDT) [reply src/locales/ro_MD/date/month.ts]: Don't remove everything, only the `_context` ones.

File: src/locales/ro_MD/date/weekday.ts [hunk 17]
```diff
@@ -0,0 +1,10 @@
+export default {
+  wide: [],
```
Comment (matthewmayer) [inline src/locales/ro_MD/date/weekday.ts]: Sorry not what I meant. You need to provide the lists of months/days for the wide and abbr keys but you don't need the wide_context and abbr_context keys at all

**Review by matthewmayer — CHANGES_REQUESTED** [review]

**Comment by matthewmayer** [comment]
Btw recommended you don't force push, that way it's easier to see the history of the PR. It gets squash-merged at the end anyway!

File: src/locales/ro_MD/date/weekday.ts (hunk 17 above)
Comment (botzill) [reply src/locales/ro_MD/date/weekday.ts]: I see, I did copy it from `ro`, seems that it needs to be changed in `ro` as well.

File: src/locales/ro_MD/date/weekday.ts [hunk 18]
```diff
@@ -0,0 +1,4 @@
+export default {
+  wide: ['Luni', 'Marți', 'Miercuri', 'Joi', 'Vineri', 'Sâmbătă', 'Duminică'],
+  abbr: ['Luni', 'Marți', 'Miercuri', 'Joi', 'Vineri', 'Sâmbătă', 'Duminică'],
```
Comment (ST-DDT) [inline src/locales/ro_MD/date/weekday.ts]: The abbreviations are usually 2 or three characters.
Are you sure there aren't any abbreviations available?

File: src/locales/ro_MD/location/street.ts [hunk 19]
```diff
@@ -0,0 +1 @@
+export default ['{{location.street_prefix}} {{location.street_name}}'];
```
Comment (ST-DDT) [inline src/locales/ro_MD/location/street.ts]: Usually `location.street_name` refers to existing street names, maybe rename the file to `location.street_name_part`?
Any alternative suggestions?

File: src/locales/ro_MD/location/county.ts [hunk 20]
```diff
@@ -0,0 +1,36 @@
+export default [
```
Comment (ST-DDT) [inline src/locales/ro_MD/location/county.ts]: This is the same as state. Is that intended?

Comment (botzill) [reply src/locales/ro_MD/location/county.ts]: I removed the states as in MD there are not state and let only county

File: src/locales/ro_MD/location/state_abbr.ts [hunk 21]
```diff
@@ -0,0 +1,36 @@
+export default [
+  'AN',
```
Comment (ST-DDT) [inline src/locales/ro_MD/location/state_abbr.ts]: I assume these are now also wrong?

Comment (botzill) [reply src/locales/ro_MD/location/state_abbr.ts]: yes fixed now

**Comment by ST-DDT** [comment]
I'm almost done with reviewing this, I will do the rest this weekend.
Thanks for your contribution and your quick fixes ❤️ .

**Comment by botzill** [comment]
> I'm almost done with reviewing this, I will do the rest this weekend. Thanks for your contribution and your quick fixes ❤️ .

Thanks

File: src/locales/ro_MD/phone_number/formats.ts [hunk 22]
```diff
@@ -0,0 +1,12 @@
...
+  '0220#####',
+  '0221#####',
+  '0222#####',
+  '0223#####',
+  '0224#####',
+  '0225#####',
+  '0226#####',
+  '0227#####',
+  '0228#####',
+  '0229#####',
```
Comment (ST-DDT) [inline src/locales/ro_MD/phone_number/formats.ts]: According to [wikipedia](https://en.wikipedia.org/wiki/Telephone_numbers_in_Moldova), 3 is also used for land line telephone numbers. Or just on paper?

Comment (botzill) [reply src/locales/ro_MD/phone_number/formats.ts]: This is actually the phone code only for the capital, I did not include other locations.

Comment (botzill) [reply src/locales/ro_MD/phone_number/formats.ts]: @ST-DDT I think is OK to generate only this numbers, as are most common ones.

**Comment by botzill** [comment]
Hi @ST-DDT, let me know, please, if there are any other changes required. 

Thanks.

**Review by Shinigami92 — DISMISSED** [review]

**Comment by botzill** [comment]
How should I fix this:

```
❯ test/all_functional.spec.ts  (30506 tests | 6 failed) 7505ms
   ❯ test/all_functional.spec.ts > functional tests > ro_MD > location > state()
     → expected undefined to be truthy
   ❯ test/all_functional.spec.ts > functional tests > ro_MD > location > stateAbbr()
     → expected undefined to be truthy
   ❯ test/all_functional.spec.ts > functional tests > ro_MD > person > suffix()
     → expected undefined to be truthy
   ❯ test/all_functional.spec.ts > faker.helpers.fake functional tests > ro_MD > location > state()
     → expected 'undefined' not to be 'undefined' // Object.is equality
   ❯ test/all_functional.spec.ts > faker.helpers.fake functional tests > ro_MD > location > stateAbbr()
     → expected 'undefined' not to be 'undefined' // Object.is equality
   ❯ test/all_functional.spec.ts > faker.helpers.fake functional tests > ro_MD > person > suffix()
     → expected 'undefined' not to be 'undefined' // Object.is equality
```

?

**Comment by Shinigami92** [comment]
I assume this test ensures that inside some locale files exporting empty arrays is not allowed
Sadly the test result message is not indicating this in any way

@ST-DDT please tell me if I'm right or wrong

**Comment by ST-DDT** [comment]
> How should I fix this

For now please add them to the skipped locale entries (with a comment that this locale doesn't have that data).

> I assume this test ensures that inside some locale files exporting empty arrays is not allowed
> Sadly the test result message is not indicating this in any way

That is correct. It will do so when https://github.com/faker-js/faker/issues/893 is merged.

**Comment by botzill** [comment]
> For now please add them to the skipped locale entries (with a comment that this locale doesn't have that data).

Hi, can you guid on how todo this?

**Review by ST-DDT — APPROVED** [review]

**Comment by ST-DDT** [comment]
> > For now please add them to the skipped locale entries (with a comment that this locale doesn't have that data).
> 
> Hi, can you guid on how todo this?

Done. Thanks for your patience.

**Review by matthewmayer — CHANGES_REQUESTED** [review]
Now that #1340 has been closed we have support for multiple fallback languages, so it would make sense to change the fallback languages

ro_MD -> ro -> en, so ro_MD only includes MD-specific data.

**Review by xDivisionByZerox — APPROVED** [review]

**Comment by ST-DDT** [comment]
@botzill Did you close this PR by accident?

**Comment by botzill** [comment]
hm, @ST-DDT I think yes,

**Comment by botzill** [comment]
How can I get it back?

**Comment by ST-DDT** [comment]
The force push comment contains a reference to the old version: https://github.com/faker-js/faker/commit/91a0aa8e271738596841b4b20718a56b2645c651

You should be able to locally checkout the commit or reset your branch to that commit.

**Comment by botzill** [comment]
Hi @ST-DDT, I do not have that comit on my end. I did rebase and I create a new PR, let me know if that helps: #1936
